cfgmgr/intfmgr.cpp

```diff
@@ -74,6 +72,87 @@ void IntfMgr::setIntfVrf(const string &alias, const string vrfName)
     EXEC_WITH_ERROR_THROW(cmd.str(), res);
 }
 
+void IntfMgr::addHostSubIntf(const string&intf, const string &subIntf, const string &vlan)
+{
+    size_t found = subIntf.find(VLAN_SUB_INTERFACE_SEPARATOR);
+    if (found == string::npos)
+    {
+        SWSS_LOG_ERROR("Invalid host sub interface name: %s", subIntf.c_str());
+        return;
+    }
+    if (intf != subIntf.substr(0, found))
```

Comment:
`intf `and vlan are extracted from `subIntf ` in doTask and then you are doing an extra check? Looks redundant to me.

Reply:
removed